# Задание 1

Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

In [78]:
import math
import numpy as np

In [79]:
std = 16
a = 0.95
M = 80
n = 256

# Стандартное отклонение генеральной совокупности известно, поэтому воспользуемся Z-критерием.
# Формула: X +- Z(a/2) * std / sqrt(n)

Z = 1.96
right_boarder = M + (1.96 * std / math.sqrt(256))
left_boarder = M - (1.96 * std / math.sqrt(256))

print('С вероятностью 95% среднее генеральной совокупности лежит в интервале от {} до {}.'.format(left_boarder, right_boarder))

С вероятностью 95% среднее генеральной совокупности лежит в интервале от 78.04 до 81.96.


# Задание 2

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95. 

In [83]:
data = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

In [93]:
def mean(a):
    return round(a.sum() / len(a), 2)

In [94]:
def std(x):
    return round(math.sqrt(np.sum(np.power(x - mean(x),2)) / (len(x)-1)), 2)

In [86]:
# посчитаем среднее значение и стандартное отклонение
m = mean(data)
n = len(data)
std = std(data)

# среднеквадратическое отклонение генеральной совокупности не известно, решаем через t-критерий.
# степеней свободы = 9, так как наблюдений 10.

t = 2.262


right_boarder = round(m + (2.262 * std / math.sqrt(n)), 2)
left_boarder = round(m - (2.262 * std / math.sqrt(n)), 2)

print('Истинное значение среднего генеральной совокупности лежит в диапазоне от {} до {}'.format(left_boarder, right_boarder))

Истинное значение среднего генеральной совокупности лежит в диапазоне от 6.27 до 6.91


# Задание 3

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм.

In [89]:
M0 = 17
M1 = 17.5
n = 100
std = 2

# Нулевая гипотеза M0 = M1. Альтернативная M1 > M0
# Чтобы понять, насколько вероятно такое отклонение M0 от M1 среднего генеральной совокупности, 
# нам нужно определить, находится ли M1 в рамках возможных значений вариативности среднего генеральной совокупности с вероятностью в 95%.

# По таблице Z-критерия находим, что отклонение в бОльшую сторону среднего ген совокупности может быть не больше, чем на 1.65 сигм,
# чтобы мы могли быть уверены в том, что данное отклонение с вероятностью 95% принадлежит той же совокупности.

# Посчитаем, на сколько сигм M1 отклоняется от M0
diff = (M1 - M0) / (std / 10)

print('Оценочное среднее отклоняется от среднего генеральной совокупности на {} сигм'.format(diff))

print('Так как среднее выборки отклоняется от среднего генеральной совокупности больше, чем на 1.65 сигм, то у нас есть все основания отвергнуть нулевую гипозету и принять альтернативную')

Оценочное среднее отклоняется от среднего генеральной совокупности на 2.5 сигм
Так как среднее выборки отклоняется от среднего генеральной совокупности больше, чем на 1.65 сигм, то у нас есть все основания отвергнуть нулевую гипозету и принять альтернативную


# Задание 4

Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [95]:
M0 = 200
n = 10
data = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [96]:
# Алгоритм:
# 1 — найдем среднее выборки
# 2 — найдем значение, на сколько может отклоняться среднее выборки так, что с 99% вероятностью оно будет принадлежать той же ген совокупности
# 3 — сравним отклонение со значением выше и если оно принадлежит интервалу — значит оценочное среднее
# с 99% вероятностью принадлежит той же ген совокупности.

# так как дисперсия генеральной совокупности нам не известна — используем t-критерий.

M1 = mean(data)
std_value = std(data)

# Согласно таблице t-критерия значение может отклонятся на 2.821 сигм.
# Посчитаем, на сколько сигм отклоняется наше оценочное среднее.

diff = (M1 - M0) / (std_value / math.sqrt(9))
print(M1, diff)

198.5 -1.0112359550561798


# Ответ

Оценочное среднее отклоняется от среднего ген совокупности на 1.01 сигму, что меньше, чем возможная граница (2.821), чтобы мы были уверены в том, что с вероятностью 99% оно принадлежит той же совокупности. А значит у нас нет оснований отклонить нулевую гипозету и, утверждения продавца — верны.